### ISSUE TITLE
Steal fails to load dependencies devDependency package.json

### ISSUE THREAD

**Issue by m-mujica** [stealjs/steal#1086]
[This change](https://github.com/stealjs/steal/commit/97af89e8c8c8531c2fb0a34ab0ed1838f1be7651#diff-a42c3006320d3fabfa5ce2b5ce1121acR254) causes StealJS to try to load a devDependency package.json in the wrong path.

I can replicate consistently in this branch https://github.com/donejs/bitballs/pull/282, what I've seen so far is:

- Steal loads `public/models/user`
- `public/models/user` loads `can-set` which has `steal-qunit` as a devDependency
- Steal tries to fetch `steal-qunit`'s package.json inside `can-set` node_modules (but since node_modules is flat, it fails)

```
GET http://localhost:5000/node_modules/can-set/node_modules/steal-qunit/package.json 404 (Not Found)
```

The exact line where Steal throws is [THIS ONE](https://github.com/stealjs/steal/blob/b95bd00b618a763899b33ed23fecb4e9164a4ea0/ext/npm-crawl.js#L626)

**Comment by phillipskevin** [comment]
Any idea if this is causing https://travis-ci.org/donejs/donejs/jobs/207552130 ?

This is the error:
```
Potentially unhandled rejection [13] Error: Error loading "donejs-chat@0.0.0#index.stache!done-autorender@1.0.0-alpha.9#src/autorender" at <unknown>
Error loading "done-autorender@1.0.0-alpha.9#src/autorender" at file:/home/travis/build/donejs/donejs/donejs-chat/node_modules/done-autorender/src/autorender.js
Did not find ./node_modules/can-zone/node_modules/chai/package.json
    at FetchTask.utils.extend.next (file:/home/travis/build/donejs/donejs/donejs-chat/node_modules/steal/ext/npm-crawl.js:628:11)
```

It's only failing on Node 4.

**Comment by m-mujica** [comment]
@phillipskevin yup. Exact same error I saw in bitballs.

**Comment by matthewp** [comment]
There's a fix in https://github.com/stealjs/steal/pull/1087

**Comment by matthewp** [comment]
Fixed in 1.2.11
